### ISSUE TITLE
BUG: Proxy method for __invoke with wrong return type

### ISSUE THREAD

**Issue by robertlemke** [neos/flow-development-collection#2899]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

An original class containing an __invoke() method with a return type of `mixed` results in a auto-generated proxy method with return type `?mixed`. 

That results in the following error:

> Fatal error: Type mixed cannot be marked as nullable since mixed already includes null

### Expected Behavior

The proxy method should contain the correct return type.

### Steps To Reproduce

Create a class containing the following method:

```php
public function __invoke($objectValue, $args, $_, ResolveInfo $info): mixed
{
}
```

Protect that method with a security policy, so that a proxy method is generated.

### Environment

```markdown
- Flow: dev-master (56bd731b691270b208423d4cadf4c7c4803d71ec), shortly after the 8.1.1 release
- PHP: 8.1
```

**Comment by kdambekalns** [comment]
Woah, probably the result of Flow checking vie reflection of the return type is nullable. Having `null` in `mixed` makes that check pass, even though it must not be marked nullable. Awesome insonsistency. 🤷‍♂️

**Comment by kdambekalns** [comment]
Also, judging from

> mixed can never be optional

the `?` doesn't even mean _nullable_, but _optional_. TIL.

**Comment by robertlemke** [comment]
We need that new / revamped Reflection Service I've been talking about.

**Comment by kdambekalns** [comment]
> We need that new / revamped Reflection Service I've been talking about.

True. But that would still be a special case I would not have anticipated.
